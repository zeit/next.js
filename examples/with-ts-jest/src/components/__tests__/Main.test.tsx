import React from "react";
import { act, create, ReactTestRenderer } from "react-test-renderer";
import Main from "../Main";

describe("COMPONENT Main", () => {
  test("WILL RENDER", () => {
    let root: ReactTestRenderer;

    act(() => {
      root = create(<Main />);
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});
