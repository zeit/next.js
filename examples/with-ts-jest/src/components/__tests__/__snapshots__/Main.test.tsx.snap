// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT Main WILL RENDER 1`] = `
<main>
  <h1>
    Welcome to 
    <a
      href="https://nextjs.org"
    >
      Next.js!
    </a>
  </h1>
  <p>
    Get started by editing
     
    <code>
      pages/index.js
    </code>
  </p>
  <div>
    <a
      href="https://nextjs.org/docs"
    >
      <h2>
        Documentation →
      </h2>
      <p>
        Find in-depth information about Next.js features and API.
      </p>
    </a>
    <a
      href="https://nextjs.org/learn"
    >
      <h2>
        Learn →
      </h2>
      <p>
        Learn about Next.js in an interactive course with quizzes!
      </p>
    </a>
    <a
      href="https://github.com/vercel/next.js/tree/master/examples"
    >
      <h2>
        Examples →
      </h2>
      <p>
        Discover and deploy boilerplate example Next.js projects.
      </p>
    </a>
    <a
      href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
    >
      <h2>
        Deploy →
      </h2>
      <p>
        Instantly deploy your Next.js site to a public URL with Vercel.
      </p>
    </a>
  </div>
</main>
`;
